app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -101,22 +133,26 @@ def parse_az(az)
       end
 
       def parse_instance(instance)
-        uid         = "#{@subscription_id}\\#{instance.fetch_path('resourceGroup')}\\#{instance.fetch_path('name')}"
+        uid = resource_uid(@subscription_id,
+                           instance.fetch_path('resourceGroup'),
+                           instance.fetch_path('type').downcase,
+                           instance.fetch_path('name'))
```

Comment:
@bronaghs The `uid` is used to fetch path in `@data`. What you want to change is probably the `ems_ref` which can be different from `uid`, and best to be a native value from Azure. We can change the way to compose `uid`, as long as it can be correctly reconstructed at the resource parsing stage for the deployment. We are relying on the same id to make the link from instance to stack. 

The type string is <provider name>/<resource type>. For instances, it is 'microsoft.compute/virtualmachine'.